evolver/base.py

```diff
@@ -92,19 +92,29 @@ def model_validate(cls, obj, *, strict=None, from_attributes=None, context=None)
 
 class _BaseConfig(_BaseModel):
     @classmethod
-    def get_classinfo(cls) -> str:
+    def get_classinfo(cls, strict=True) -> str:
         """The fully qualified classname for cls's container class (if one exists).
         E.g., ``Evolver.Config.get_classinfo()`` returns ``"evolver.device.Evolver"``.
-        Raises `TypeError` if parent class is not derived from BaseInterface.
+        Returns `None` if no container class exists, e.g., the container is a module.
         """
         fqn = evolver.util.fully_qualified_name(cls)
         containers = fqn.split(".")[:-1]
         container = ".".join(containers)
         container_class = evolver.util.import_string(container)
-        if issubclass(container_class, BaseInterface):
+        if not isinstance(container_class, type):
+            return
+        elif issubclass(container_class, BaseInterface):
             cls = container_class
         return f"{cls.__module__}.{cls.__qualname__}"
 
+    @pydantic.model_serializer(mode="wrap", when_used='json')
+    def to_descriptor(self, handler) -> dict:
+        serialized_by_super = handler(self)
+        class_info = self.get_classinfo()
+        if class_info is None or _is_descriptor_dict(serialized_by_super):
+            return serialized_by_super
+        return dict(classinfo=class_info, config=serialized_by_super)
```

Comment:
@amitschang This is the serializer. Most of the other changes in this PR might be temp and I just committed and pushed what I had.